spring-pulsar/src/main/java/org/springframework/pulsar/listener/DefaultPulsarMessageListenerContainer.java

```diff
@@ -133,6 +137,15 @@ public void doStop() {
 		setRunning(false);
 		this.logger.info("Pausing this consumer.");
 		this.listenerConsumer.consumer.pause();
+		if (this.receiveInProgress) {
```

Comment:
What is the implication of this interrupt if the batch receive is in the middle of a batch of records? I understand the need for this when the timeout is high - but it also feels like maybe this would be an operation we did when the user asked to "stopImmediately" vs. "stopGracefully".  Does this need to be included in this proposal or added when if we add a stopImmediately/Gracefully feature? 

Either way, a comment would help understand the purpose of the interrupt.

Reply:
Ya, we can add that in a separate PR. I will add a comment.

Reply:
https://github.com/spring-projects-experimental/spring-pulsar/issues/167